extensions/amp-base-carousel/1.0/arrow.js

```diff
@@ -25,44 +25,36 @@ import {dict} from '../../../src/utils/object';
 export function Arrow(props) {
   const {
     'customArrow': customArrow,
-    'dir': dir,
+    'by': by,
     'advance': advance,
     'disabled': disabled,
   } = props;
-  const button = customArrow
-    ? customArrow
-    : renderDefaultArrow(dict({'dir': dir}));
   const {
-    'children': children,
     'disabled': customDisabled,
     'onClick': onCustomClick,
-    ...rest
-  } = button.props;
+  } = customArrow.props;
   const isDisabled = disabled || customDisabled;
   const onClick = (e) => {
     if (onCustomClick) {
       onCustomClick(e);
     }
-    advance();
+    advance(by);
```

Comment:
Could you please explain how `advance` works from the caller to here and back? I seem to keep getting confused with it.

Reply:
`advance` is passed in from `BaseCarousel` which scrolls by adjusting the `container.scrollLeft` of the `Scroller` element. The `Arrow` by default will advance forward or backward by one slide, measured by its parent `BaseCarousel` in terms of the `container.offsetWidth`. In the future these will be customizable with attributes -- one such example is `advance-count` which can set what amount of slides the arrows increment by.

Another note: Setting `container.scrollLeft` does update the state because the `Scroller` has attached a `scrollHandler` which updates the slide state on debounced scroll. The smooth scroll triggered by `advance` is caught by this handler as well. 

This is somewhat messy still because the scroller ref is duplicated between `BaseCarousel` and its underlying `Scroller` -- when this and the `amp-base-carousel` PR are in I will send the one that uses `forwardRef` to consolidate this logic centrally in `BaseCarousel` and help with some of this confusion.